contract-examples/README.md

```diff
@@ -0,0 +1,112 @@
+# Subnet EVM Contracts
+
+## Introduction
+
+Avalanche is an open-source platform for launching decentralized applications and enterprise blockchain deployments in one interoperable, highly scalable ecosystem. Avalanche gives you complete control on both the network and application layers&mdash;helping you build anything you can imagine.
+
+The Avalanche Network is composed of many subnets and chains. Chains in subnets run with customizable virtual machines. One of these virtual machines is Subnet EVM. The Subnet EVM's API is almost identical to an Ethereum node's API. Subnet EVM brings its own features like minting native tokens via contracts, restrincting contract deployer etc. These features are presented with `Stateful Precompile Contracts`. These contracts are precompiled and deployed when they're activated.
+
+The goal of this guide is to lay out best practices regarding writing, testing and deployment of smart contracts to Avalanche's Subnet EVM. We'll be building smart contracts with development environment [Hardhat](https://hardhat.org).
+
+## Prerequisites
+
+### NodeJS and Yarn
+
+First, install the LTS (long-term support) version of [nodejs](https://nodejs.org/en). This is `16.2.0` at the time of writing. NodeJS bundles `npm`.
+
+Next, install [yarn](https://yarnpkg.com):
+
+```zsh
+npm install -g yarn
+```
+
+### Solidity and Avalanche
+
+It is also helpful to have a basic understanding of [Solidity](https://docs.soliditylang.org) and [Avalanche](https://docs.avax.network).
+
+## Dependencies
+
+Clone the repo and install the necessary packages via `yarn`.
+
+```zsh
+$ git clone https://github.com/ava-labs/subnet-evm.git
+$ cd smart-contract-examples
+$ yarn
+```
+
+## Write Contracts
+
+`ERC20NativeMinter.sol` is an [Open Zeppelin](https://openzeppelin.com) [ERC20](https://eips.ethereum.org/EIPS/eip-20) contract powered by native minting capabilities of Subnet EVM. ERC20 is a popular smart contract interface. It uses `INativeMinter` interface to interact with `NativeMinter` precompile.
```

Comment:
nit: it seems like this says this is an OpenZeppelin contract. Can we clarify that it is based on an OpenZeppelin contract with added capabilities?